tests/series/test_cast.py

```diff
@@ -813,3 +814,18 @@ def test_cast_timestamp_to_time_unsupported_timeunit(timeunit):
     input = Series.from_pylist([datetime(2022, 1, 6, 12, 34, 56, 78)])
     with pytest.raises(ValueError):
         input.cast(DataType.time(timeunit))
+
+
+def test_cast_binary_to_fixed_size_binary():
+    data = [b"abc", b"def", None, b"bcd", None]
+
+    input = Series.from_pylist(data)
+    assert input.datatype() == DataType.binary()
+    casted = input.cast(DataType.fixed_size_binary(3))
+    assert casted.to_pylist() == [b"abc", b"def", None, b"bcd", None]
+
+    data.append(b"long")
```

Comment:
For better testing clarity, make this into a new test: `test_cast_binary_to_fixed_size_binary_fails_with_variable_lengths` 

It's generally cleaner to have each test only test a single pattern, and also much easier for debugging!